satpy/tests/reader_tests/test_goes_nc.py

```diff
@@ -0,0 +1,511 @@
+                coefs['a'].append(self._float(row[2]))
+                coefs['b'].append(self._float(row[3]))
+
+        return coefs
+
+    def _get_vis_coefs(self, platform):
+        # Find calibration table
+        table = self._get_table(root=self.vis_html,
+                                heading=self.vis_tables[platform],
+                                heading_type='p')
+
+        # Extract values
+        coefs = defaultdict(list)
+        if platform in ('GOES-8', 'GOES-9'):
+            # GOES 8&9 coefficients are in the same table
+            col = 1 if platform == 'GOES-8' else 2
+            coefs['slope'].append(self._float(table[1][col]))
+            coefs['x0'] = self._float(table[2][col])
+            coefs['offset'].append(self._float(table[3][col]))
+            coefs['k'] = self._float(table[4][col])
+        else:
+            # k and x0 appear in the first row only
+            coefs['slope'].append(self._float(table[0][1]))
+            coefs['x0'] = self._float(table[0][2])
+            coefs['k'] = self._float(table[0][4])
+            coefs['offset'].append(self._float(table[0][3]))
+
+            # Remaining rows
+            for row in table[1:]:
+                coefs['slope'].append(self._float(row[1]))
+                coefs['offset'].append(self._float(row[2]))
+
+        return coefs
+
+    def _get_table(self, root, heading, heading_type, ):
+        # Find table by its heading
+        headings = [h for h in root.find_all(heading_type)
+                    if heading in h.text]
+        if not headings:
+            raise ValueError('Cannot find a coefficient table matching text '
+                             '"{}"'.format(heading))
+        elif len(headings) > 1:
+            raise ValueError('Found multiple headings matching text "{}"'
+                             .format(heading))
+        table = headings[0].next_sibling.next_sibling
+
+        # Copy items to a list of lists
+        tab = list()
+        for row in table.find_all('tr'):
+            cols = row.find_all('td')
+            if cols:
+                tab.append([c.text for c in cols])
+        return tab
+
+    def _denoise(self, string):
+        return string.replace('\n', '').replace(' ', '')
+
+    def _float(self, string):
+        """Convert string to float
+
+        Take care of numbers in exponential format
+        """
+        string = self._denoise(string)
+        exp_match = re.match('^[-.\d]+x10-(\d)$', string)
+        if exp_match:
+            exp = int(exp_match.groups()[0])
+            fac = 10 ** -exp
+            string = string.replace('x10-{}'.format(exp), '')
+        else:
+            fac = 1
+
+        return fac * float(string)
+
+
+class GOESNCFileHandlerTest(unittest.TestCase):
+
+    longMessage = True
+
+    @mock.patch('satpy.readers.goes_nc.xr')
+    def setUp(self, xr_):
+        # Disable logging
+        logging_off()
+        logging.getLogger("urllib3").setLevel(logging.WARNING)
```

Comment:
You turn logging_off here but never turn it back on, right? Why do you need to turn it off? Isn't setLevel for urllib3/bs4 good enough?

Reply:
Another good catch. I forgot to remove that line.